src/lib_file.cc

```diff
@@ -336,6 +336,24 @@ SString GetAbsolutePath(const SString &path)
 	return stringBuffer.data();
 }
 
+// Thu 12 May 2022
+// Assuming fl_filename_relative can be safely wrapped as above
+//
+SString GetRelativePath(const SString &path)
+{
+        size_t sz = 64;
+        std::vector<char> stringBuffer;
+
+        do
+        {
+                sz *= 2;
+                stringBuffer.resize(sz);
+                fl_filename_relative(stringBuffer.data(), (int)stringBuffer.size(), path.c_str());
+        } while(stringBuffer.back() == '\0' && stringBuffer[stringBuffer.size() - 2] != 0);
```

Comment:
I think we should just use [`FL_PATH_MAX`](https://www.fltk.org/doc-1.3/group__filenames.html#ga29253083ed9918c2f9d532bd5ed89cc3) fixed-length arrays for the FLTK file path functions. Your safety measure above is good, but it's acceptable to just assume `FL_PATH_MAX`.

Reply:
So to be clear, both the relative and absolute path functions should be rewritten, simplified, with buffers of FL_PATH_MAX length, then?

Reply:
Ah, alright, nevermind then, guess I didn't follow tradition anyway. Then it's good to be as it is.